crates/rust-analyzer/src/handlers.rs

```diff
@@ -1052,17 +1058,35 @@ pub(crate) fn handle_code_action_resolve(
         .only
         .map(|it| it.into_iter().filter_map(from_proto::assist_kind).collect());
 
+    let assist_kind: AssistKind = match params.kind.parse() {
+        Ok(kind) => kind,
+        Err(e) => {
+            return Err(LspError::new(
+                ErrorCode::InvalidParams as i32,
+                format!("For the assist to resolve, failed to parse the kind: {}", e),
+            )
+            .into())
+        }
+    };
+
     let assists = snap.analysis.assists_with_fixes(
         &assists_config,
         &snap.config.diagnostics(),
-        true,
+        AssistResolveStrategy::Single(SingleResolve { assist_id: params.id.clone(), assist_kind }),
         frange,
     )?;
 
-    let (id, index) = split_once(&params.id, ':').unwrap();
-    let index = index.parse::<usize>().unwrap();
-    let assist = &assists[index];
-    assert!(assist.id.0 == id);
+    let assist = &assists[params.index];
```

Comment:
This probably should use `.get` to avoid panic. Technically, that's untrusted user input and all that.